### ISSUE TITLE
[Feature Request] Add an accounts carousel / ribbon in Transactions tab too. 

### ISSUE THREAD

**Issue by bikram-agarwal** [jameskokoska/Cashew#105]
**Is your feature request related to a problem? Please describe.**
When I'm at the "Transactions" tab, almost always I need to see the transactions from some specific account. So, right now, these are the steps, after going to transactions tab - 
1. Click the "filters" icon. 
2. Scroll the account list and click the account I want. 

![Filter](https://github.com/jameskokoska/Cashew/assets/4424058/693f25da-e215-45ab-af63-2419c225b04c)

Problem # 1 -> the filter window is kinda small. I'm using the webapp in a 4K screen. The webapp is 3840 pixels wide, but the filter screen is only 915 pixels wide. I can only see 5 accounts at a time here. With my 22 accounts, I have to scroll quite a bit. 

Problem # 2 -> I can't see any account's total balance in the transactions tab. If my suggested feature is implemented, users will be able to see account's total balance in the transactions tab too.

**Describe the solution you'd like**
The accounts carousel / ribbon that is currently visible in Home tab, please add a similar carousel in Transactions tab too. Tapping any of the accounts in this carousel will filter transactions for that account. 

![Modified Transactions](https://github.com/jameskokoska/Cashew/assets/4424058/b6d12e5f-b010-4a85-a36d-270ce10b1db8)

That way, the account selector ribbon would be 3420 pixels wide. I can see TEN of my accounts at a time. I won't have to scroll most of the time to select the account I want to see. 
Also, I can see each account's balance in the ribbon, in this transactions tab too. Two birds with one stone!

**Comment by jameskokoska** [comment]
It's an interesting idea, personally it's nice to also see all the transactions instead of always one account being selected in the transactions tab. I think it also creates confusion between an account selection and a filters option. 

Do you have any ideas how to address these issues?

**Comment by bagarwa** [comment]
> It's an interesting idea, personally it's nice to also see all the transactions instead of always one account being selected in the transactions tab. I think it also creates confusion between an account selection and a filters option.
> 
> Do you have any ideas how to address these issues?

You can add a "All" box on the leftmost side, before all other accounts' boxes. That way, user can tap "All" and see all accounts' transactions. 

Each of the account cards in the carousel will act same as account chips in the Filters window. That is, users can tap / click to select multiple accounts from the accounts carousel. 

# Example

Say I tap "Citi Checking", "Citi Savings" and "Citi CD" cards from the carousel. Those 3 cards will be highlighted as selected. AND those three account chips will also show up in the selected filter list. Like so - 
![Modified Transactions](https://github.com/jameskokoska/Cashew/assets/4424058/b6db5cb0-6be6-4108-8ffb-d1718041b3be)

If I tap "Citi CD" card in the carousal again, that account will be unselected and its chip will also be removed from that selected filter list. 
Alternatively, if the user taps the filter chip itself, and then unselects "Citi CD" in the filter window, "Citi CD" will automatically be unselected from the carousal too.

**Comment by jameskokoska** [comment]
An all account selector definitely makes sense however the account selector proposed is the same as the one on the homepage. Tapping an account on the homepage currently sets the primary account and currency. While tapping an account in the new proposed switcher will select the accounts to show transactions from. This will create a lot of confusion for users. Any ideas on how to solve this issue?

Whereas the current implementation through filters uses an entirely different known selection for selecting which accounts to show up on that page.

**Comment by jameskokoska** [comment]
To add on to this, the transactions list isn't meant to be used as an in-depth analysis of transaction spending. The All Spending page is better suited to have a feature such as an instant account switcher, especially since the time period can be customized there. Is there a reason why the feature is proposed for the transactions list page?

Then to add on to that, the individual account can be tapped on the homepage to view a breakdown of the spending directly related to that account. This directly eliminates the need for any quick account selection as this feature is already in the app.


Separately:

Maybe a better feature would be to use the empty space on the side to be able to 'pin' the available filters for quick access in full screen mode.

**Comment by bagarwa** [comment]
> Tapping an account on the homepage currently sets the primary account and currency. 

Oh, so that's what it is doing!! Before you told me, I didn't even understand what tapping that once does; as nothing else on the home page was changing. Tapping it second time opens that account's transaction analysis etc. 

If the purpose of the accounts carousel in Home page is letting user select the primary account, then maybe instead of carousal, you can make it a dropdown list. Only the primary account will be visible, and clicking the dropdown icon next to it will open a vertical list of all accounts and user can click the account they want to set as primary account. The list will close and show only that selected account now. 

<img width="1255" alt="Homescreen" src="https://github.com/jameskokoska/Cashew/assets/4424058/07536d2f-2f00-4005-97e4-ef3ef19f7319">

> the transactions list isn't meant to be used as an in-depth analysis of transaction spending. 

My feature request doesn't do that though. It's for easier access to account switcher (less scrolling, less tapping) and total account balance visibility in transaction page.

**Comment by jameskokoska** [comment]
I think a dropdown is a bit complicated of a user interaction for the homepage, especially given a mobile layout - which is the main focus of Cashew.

I see that the main focus of this suggestion is a quick access account switcher, but you can specifically view the details of an account by selecting the account in the homepage once selected as the primary. On the account details page it will show the total for the account and breakdown of spending along with transactions.

Because of this reason, I'm starting to think the need for an account switcher showing specific transactions on the transaction page is redundant. And can be accomplished by filters if needed.

I understand with many accounts filters may be difficult to apply, but if you are just interested in the details of one account at a time, then the account details page can provide you this information (albeit a double click after the account is selected as the primary)

**Comment by ngocanhtve** [comment]
I see a feature very similar to this request's description exists in Money Lover.

**Comment by jameskokoska** [comment]
I understand other apps may have features, however, Cashew is a different application and usability and functionality needs to be considered from Cashew's base functions first.

Going to close this issue for now since it seems redundant. Main reason being, the account details page can provide insights into an specific accounts spending (without needing to apply filters) and the summary of account spending can be found on the homepage already.

Glad to continue the conversation though

**Comment by bagarwa** [comment]
> I see a feature very similar to this request's description exists in Money Lover.

Yep, that's what I've been using since years and it's perfect for me. I have the premium version. Only reason I'm now trying to find an alternative is that they closed their web app. Cashew is the best alternative I found. Its UI is great and it works great; other than a few feature parity. In MoneyLover, I can do all activities I need in the transactions tab itself. But here I have to constantly switch between different tabs and views. I understand there's pros and cons of both approaches. But years of familiarity is hard to get rid of. :D

**Comment by ngocanhtve** [comment]
> however, Cashew is a different application and usability and functionality needs to be considered from Cashew's base functions first.

Of course. Cashew is a great app but also a new one. Long-standing apps like Money Lover have implemented several features that can win them loyal customers. So it's understandable when users ask Cashew to have features that users were familiar with before. So either Cashew has a solution for the problem that users need. Or Cashew must re-implement solutions that already exist on long-standing applications like Money Lover.

I would also like to add that I have not used Money Lover before. I'm just giving my opinion. I feel Cashew's filter meets the need well and solves the same problem as Money Lover have implemented. I am satisfied with Cashew's filter. But you can still make it better, right? I believe in you.

**Comment by jameskokoska** [comment]
This has been added in the latest beta. Account filters can be pinned to the top of the transactions list on the Monthly Transactions page. To enable this feature, tap the three dot menu in the top right corner on the Monthly Transactions page, and enable Pin Account Filters.
See #725 for Beta Testing instructions and information.
